src/Model/Locale.php

```diff
@@ -296,4 +301,47 @@ public function getChain()
 
         return $this->chain;
     }
+
+    /**
+     * Fetch a native language string from the {@link i18n} class via the current locale code in the format "XX_xx". In
+     * the event a match cannot be found in any framework resource, an empty string is returned.
+     *
+     * @return string The native language string for the current locale e.g. "portugu&ecirc;s (Brazil)"
+     */
+    public function getNativeName()
+    {
+        $locales = i18n::getData();
+
+        // Attempts to fetch the native language string via the `i18n::$common_languages` array
+        if ($native = $locales->languageName($locales->langFromLocale($this->Locale))) {
+            return $native;
+        }
+
+        return '';
+    }
+
+    /**
+     * Determine the base URL within the current locale
+     *
+     * @return string
+     */
+    public function getBaseURL()
+    {
+        $base = Director::baseURL();
+
+        // Build domain-specific base URL
+        if ($this->Domain() && $this->Domain()->exists()) {
```

Comment:
We should only do this in domain mode. Even if a domain is configured (but not in domain mode) we should default to normal locale-prefix baseurl.